src/main/java/com/faforever/client/game/CreateGameController.java

```diff
@@ -375,23 +376,39 @@ public void onRandomMapButtonClicked() {
     mapListView.scrollTo(mapIndex);
   }
 
-  public void onGenerateMapButtonClicked() {
+  public void onGenerateMapButtonClicked(){
+    onGenerateMap();
+  }
+
+  private void onGenerateMap() {
     try {
-      mapGeneratorService.generateMap().thenAccept(mapName -> {
-        Platform.runLater(() -> {
-          initMapSelection();
-          mapListView.getItems().stream()
-              .filter(mapBean -> mapBean.getFolderName().equalsIgnoreCase(mapName))
-              .findAny().ifPresent(mapBean -> {
-            mapListView.getSelectionModel().select(mapBean);
-            mapListView.scrollTo(mapBean);
-            setSelectedMap(mapBean);
+      mapGeneratorService.setGeneratorVersion(mapGeneratorService.queryMaxSupportedVersion());
+      if (mapGeneratorService.getGeneratorVersion().compareTo(new ComparableVersion("1")) < 0) {
```

Comment:
The hardcoded "1" needs at least a comment what is happening here.

Reply:
Done